kafka.cpp

```diff
@@ -36,6 +38,31 @@ static void dr_msg_cb(rd_kafka_t *rk, const rd_kafka_message_t *rkmessage, void
 	}
 }
 
+/**
+ *  Callback to handle errors
+ */
+static void error_cb(rd_kafka_t *rk, int err, const char *reason, void *opaque)
+{
+    rd_kafka_resp_err_t kafkaError = (rd_kafka_resp_err_t) err;
+	switch (kafkaError)
+	{
+		case RD_KAFKA_RESP_ERR__TRANSPORT:
```

Comment:
You can probably combine some of the cases here.

Reply:
Fixed

Reply:
I think Aman meant to combine the "all brokers down" and "transport" errors cases, as they have same statements execution.